pyprecag/processing.py

```diff
@@ -2537,3 +2401,222 @@ def ttest_analysis(points_geodataframe, points_crs, values_raster, out_folder,
+                    UPPER category
+        upper_percentage (int): the percent variation either above/below the mean to apply
+                    to the UPPER raster category.
+        upper_probability (int):the probability percentage to apply to the LOWER category
+
+        lower_raster_files (List[str]): List of rasters to used for the analysis of the
+                    LOWER category
+        lower_percentage (int): the percent variation either above/below the mean to apply
+                    to the LOWER raster category.
+        lower_probability (int): the probability percentage to apply to the LOWER category
+
+        output_tif (str): Output TIF file
+
+    """
+
+    for arg_check in [('upper_raster_files', upper_raster_files),
+                      ('lower_raster_files', lower_raster_files)]:
+        if not isinstance(arg_check[-1], list):
+            raise TypeError('Invalid Type: {} should be a list'.format(arg_check[0]))
+
+        if len(arg_check[-1]) == 0:
+            raise TypeError('Invalid Type: {} is a empty list'.format(arg_check[0]))
+
+    for arg_check in [('upper_percentage', upper_percentage),
+                      ('upper_probability', upper_probability),
+                      ('lower_percentage', lower_percentage),
+                      ('lower_probability', lower_probability)]:
+        if not isinstance(arg_check[1], int):
+            raise TypeError('{} must an integer'.format(arg_check[0]))
+
+    for arg_check in [('upper_percentage', upper_percentage),
+                      ('lower_percentage', lower_percentage)]:
+        if not isinstance(arg_check[1], int):
+            raise ValueError('{} must an integer between -50 to 50.'.format(arg_check[0]))
+
+    if output_tif is None or output_tif == '':
+        raise TypeError('Please specify an output filename')
+
+    if not os.path.exists(os.path.dirname(output_tif)):
+        raise IOError('Output directory {} does not exist'.format(os.path.dirname(output_tif)))
+
+    start_time = time.time()
+    temp_file_list = []
+    out_prefix, out_ext = os.path.splitext(os.path.basename(output_tif))
+
+    # Process upper ------------------------------------------------------------------------------
+    temp_file_list = [os.path.join(TEMPDIR, out_prefix + '_1_upperyears' + out_ext)]
+    upper_tif = persistor_all_years(upper_raster_files,
+                                    temp_file_list[-1],
+                                    greater_than=True,
+                                    target_percentage=upper_percentage)
+
+    upper_cutoff = (upper_probability / 100.00) * len(upper_raster_files)
+    with rasterio.open(upper_tif, 'r') as src:
+        kwargs = src.meta.copy()
+        band = src.read(masked=True)
+
+        # Greater than test = retuns boolean which can be converted to 0,1
+        data_u = np.where(~band.mask, (band >= upper_cutoff).astype(np.int16), -9999)
```

Comment:
Change ">=" to >

Reply:
The original code uses the GreaterThanEqual tool for both upper and lower.

Reply:
fixed